shapash/explainer/smart_plotter.py

```diff
@@ -2323,3 +2342,474 @@ def correlations(
+        max_features : int, optional
+            Maximum number of displayed features, by default 10
+        file_name: string, optional
+            Specify the save path of html files. If it is not provided, no file will be saved, by default None
+        auto_open: bool, optional
+            open automatically the plot, by default False
+
+        Returns
+        -------
+        fig
+            The figure that will be displayed
+        """
+        assert index in self.explainer.x_pred.index, "index must exist in pandas dataframe"
+
+        self.explainer.compute_features_stability([index])
+
+        column_names = np.array([self.explainer.features_dict.get(x) for x in self.explainer.x_pred.columns])
+        ordinal = lambda n: "%d%s" % (n, "tsnrhtdd"[(math.floor(n / 10) % 10 != 1) * (n % 10 < 4) * n % 10:: 4])
+
+        # Compute explanations for instance and neighbors
+        g = self.explainer.local_neighbors["norm_shap"]
+
+        # Reorder indices based on absolute values of the 1st row (i.e. the instance) in descending order
+        inds = np.flip(np.abs(g[0, :]).argsort())
+        g = g[:, inds]
+        columns = [column_names[i] for i in inds]
+
+        # Plot
+        g_df = pd.DataFrame(g, columns=columns).T.rename(
+                columns={
+                    **{0: "instance", 1: "closest neighbor"},
+                    **{i: ordinal(i) + " closest neighbor" for i in range(2, len(g))},
+                }
+            )
+
+        # Keep only max_features
+        if max_features is not None: g_df = g_df[:max_features]
+
+        fig = go.Figure(data=[go.Bar(name=g_df.iloc[::-1, ::-1].columns[i],
+                        y=g_df.iloc[::-1, ::-1].index.tolist(),
+                        x=g_df.iloc[::-1, ::-1].iloc[:, i],
+                        marker_color=self.dict_stability_bar_colors[1] if i == g_df.shape[1]-1
+                        else self.dict_stability_bar_colors[0],
+                        orientation='h',
+                        opacity=np.clip(0.2+i*(1-0.2)/(g_df.shape[1]-1), 0.2, 1)) for i in range(g_df.shape[1])])
+
+        title = "Comparing explanations in a neighborhood:"
+        title += "<span style='font-size: 16px;'><br />How similar are explanations for closeby neighbours?</span>"
+        dict_t = copy.deepcopy(self.dict_title_stability)
+        dict_xaxis = copy.deepcopy(self.dict_xaxis)
+        dict_yaxis = copy.deepcopy(self.dict_yaxis)
+        dict_xaxis['text'] = "Normalized SHAP value"
+        dict_yaxis['text'] = ""
+        dict_t['text'] = title
+
+        fig.update_layout(template="none",
+                          title=dict_t,
+                          xaxis_title=dict_xaxis,
+                          yaxis_title=dict_yaxis,
+                          hovermode='closest',
+                          barmode="group",
+                          height=10*g_df.shape[0]*g_df.shape[1],
```

Comment:
The height of plot seems small when:
- the number of features is smaller than 6
- small number of neighbors
![Capture d’écran 2021-09-07 à 16 14 11](https://user-images.githubusercontent.com/41585787/132359921-c2274237-4174-4a9d-a031-252b0073c874.png)